rzup/src/cmd/toolchain.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2024 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use clap::Subcommand;
+
+use crate::toolchain::{build::BuildToolchain, install::InstallToolchain, repo::ToolchainRepo};
+
+use super::show::print_installed_toolchains;
+
+#[derive(Debug, Subcommand)]
+pub enum ToolchainSubcmd {
+    /// List installed toolchains
+    List {
+        #[arg(short, long)]
+        verbose: bool,
+    },
+
+    /// Install or update a given toolchain
+    Install {
+        #[command(subcommand)]
+        subcmd: ToolchainInstallSubcmd,
+    },
+
+    /// Build the toolchain
+    Build(BuildToolchain),
+
+    /// Update toolchain
+    Update {
+        #[command(subcommand)]
+        subcmd: ToolchainInstallSubcmd,
+    },
+}
+
+#[derive(Debug, Subcommand)]
+pub enum ToolchainInstallSubcmd {
+    /// Install Rust toolchain
+    Rust {
+        /// Toolchain version (i.e. 'latest' or a specific version)
+        #[arg(required = false)]
+        toolchain: Option<String>,
+    },
+    /// Install C++ toolchain
+    Cpp {
+        /// Toolchain version
+        #[arg(required = false)]
+        toolchain: Option<String>,
+    },
```

Comment:
Opinionated: Noting that going to install this is a bit unintuitive and ideally there is just a way to install both (esp if both have prebuilt for the architecture?)

by comparison, the current `rzup` command by itself would install both to latest, where going this flow, you'd have to run `rzup toolchain install rust` and `rzup toolchain install cpp`, which seems like it could have a simple to use default for those who don't know what they need to get running.

Edit: I see now that this is matching the rustup interfaces, with some indirect comparisons (release channel swapped for rust/cpp toolchain), so this comment is very opinionated.

Reply:
Good call out - @SchmErik and I were just talking about this. We are probably going to go with the default `install` installing both the c and rust toolchains. imo having the specific control for toolchains selection will still be useful for more lang support in the future, though.